Firestore/core/test/firebase/firestore/immutable/array_sorted_map_test.cc

```diff
@@ -43,6 +43,12 @@ TEST(ArraySortedMap, ChecksSize) {
   ASSERT_ANY_THROW(map.insert(next, next));
 }
 
+TEST(ArraySortedMap, InitializerIsSorted) {
+  IntMap map{{3, 0}, {2, 0}, {1, 0}};
+
+  EXPECT_EQ(map.begin()->first, 1);
```

Comment:
Very optional: you could use [`std::is_sorted`](https://en.cppreference.com/w/cpp/algorithm/is_sorted). I think `std::pair` should have a working `operator<` if its member have one.

Reply:
Good idea; done.